packages/react-events/src/Swipe.js

```diff
@@ -159,26 +236,20 @@ const SwipeResponder = {
           }
           const x = (obj: any).screenX;
           const y = (obj: any).screenY;
-          if (x < state.x && props.onSwipeLeft) {
-            state.direction = 3;
-          } else if (x > state.x && props.onSwipeRight) {
-            state.direction = 1;
+          const distX = x - state.startX;
+          const distY = y - state.startY;
+
+          if (Math.abs(distX) >= DEFAULT_TRESHOLD_SWIP) {
+            state.direction = distX < 0 ? 'left' : 'right';
+          } else if (Math.abs(distY) >= DEFAULT_TRESHOLD_SWIP) {
+            state.direction = distY < 0 ? 'down' : 'up';
```

Comment:
currently, this logic "prefers" left/right swipes over up/down swipes?
Could/Should we check which abs is greater to determine the direction?
```
const absX = Math.abs(distX);
const absY = Math.abs(distY);
if (absX > DEFAULT_TRESHOLD_SWIP || absY > DEFAULT_TRESHOLD_SWIP) {
  if (absX > absY) {
    state.direction = distX < 0 ? 'left' : 'right';
  }
  state.direction = distY < 0 ? 'down' : 'up';
}
```